src/server/router.ts

```diff
@@ -44,22 +49,27 @@ function bundleAssetUrl(path: string) {
  * Returns a router that contains all fresh routes. Should be mounted at
  * constants.INTERNAL_PREFIX
  */
-function internalRouter(bundler: Bundler): oak.Router {
-  const router = new oak.Router();
-
-  router.get(`${JS_PREFIX}/${BUILD_ID}/:path*`, async (ctx) => {
-    const path = `/${ctx.params.path}`;
+function internalRoute(bundler: Bundler): router.MatchHandler {
+  return async (_req, match) => {
+    const path = `/${match.params.path}`;
     const file = await bundler.get(path);
+    let res;
     if (file) {
-      ctx.response.status = 200;
-      ctx.response.type = extname(path);
-      ctx.response.body = file;
-      ctx.response.headers.set(
-        "Cache-Control",
-        "public, max-age=604800, immutable",
-      );
+      const headers = new Headers({
+        "Cache-Control": "public, max-age=604800, immutable",
+      });
+
+      const contentType = mediaTypeLookup(path);
+      if (contentType) {
+        headers.set("Content-Type", contentType);
+      }
+
+      res = new Response(file, {
+        status: 200,
+        headers,
+      });
     }
-  });
 
-  return router;
+    return res ?? new Response();
```

Comment:
This will result in a 200. I want a 404 (if you don't handle a route in oak (don't modify the response) then the request will fall through to the default handler.